src/lib/ui/srv_status_component.rb

```diff
@@ -0,0 +1,229 @@
+  # Component encapsulating the widgets for managing the status of services (both
+  # currently and on system boot) and the behavior associated to those widgets
+  #
+  # As long as #handle_input is invoked in the event loop, the component will
+  # handle interactive starting and stopping of the service on user demand. In
+  # addition #reload_if_requested can be used after saving the settings.
+  #
+  # To manage the status on boot, the component can be queried for the user
+  # selection using #enabled?. In addition enabled_callback (in constructor)
+  # can be used to observe the status of the corresponding field in the UI.
+  class SrvStatusComponent
+    include Yast::UIShortcuts
+    include Yast::I18n
+    include Yast::Logger
+
+    # @param service_name [String] name of the service as expected by
+    #   Yast::Service
+    # @param reload [Boolean] initial value for the "reload" checkbox.
+    #   Keep in mind it will always be displayed as unchecked if the service is
+    #   not running, despite the real value.
+    # @param enabled_callback [Proc] callback executed when the "enabled on
+    #   boot" checkbox is changed. The only parameter of the callback is the new
+    #   state of the checkbox (boolean).
+    def initialize(service_name, reload: true, enabled_callback: nil)
+      @service_name = service_name
+      @reload = reload
+      @enabled_callback = enabled_callback
+
+      @enabled = service_enabled?
+      @id_prefix = "_srv_status_#{@service_name}"
+    end
+
+    # @return [YaST::Term]
+    def widget
+      Frame(
+        _("Service Status"),
+        VBox(
+          ReplacePoint(Id("#{id_prefix}_status"), status_widget),
+          reload_widget,
+          VSpacing(),
+          on_boot_widget
+        )
+      )
+    end
+
+    # Handles the input triggered by the widgets, this method must be called in
+    # the event loop of the dialog using the component.
+    def handle_input(input)
+      case input
+      when "#{id_prefix}_stop"
+        stop_service
+        refresh_widget
+      when "#{id_prefix}_start"
+        start_service
+        refresh_widget
+      when "#{id_prefix}_reload"
+        @reload = Yast::UI.QueryWidget(Id(input), :Value)
+      when "#{id_prefix}_enabled"
+        @enabled = Yast::UI.QueryWidget(Id(input), :Value)
+        @enabled_callback.call(@enabled) if @enabled_callback
+      end
+    end
+
+    # Updates the widget to reflect the current status of the service and the
+    # settings
+    def refresh_widget
+      Yast::UI.ChangeWidget(Id("#{id_prefix}_reload"), :Enabled, service_running?)
+      Yast::UI.ChangeWidget(Id("#{id_prefix}_reload"), :Value, service_running? && @reload)
+      Yast::UI.ChangeWidget(Id("#{id_prefix}_enabled"), :Value, @enabled)
+      Yast::UI.ReplaceWidget(Id("#{id_prefix}_status"), status_widget)
+    end
+
+    # Reloads the service if the user requested so. It should be called after
+    # saving the settings.
+    def reload_if_requested
```

Comment:
I do not like name, why not simple `reload` or `reload!` to provide hint it is force reload?